plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -175,6 +170,15 @@ scriptInstance escrow = Scripts.Validator @Escrow
 escrowAddress :: EscrowParams -> Ledger.Address
 escrowAddress = Scripts.scriptAddress . scriptInstance
 
+escrowContract :: EscrowParams -> Contract EscrowSchema ()
+escrowContract escrow = 
+    let payAndRefund = do
+            (ownPubKey, vl) <- endpoint @"pay-escrow"
+            _ <- pay escrow ownPubKey vl
+            _ <- awaitSlot (escrowDeadline escrow)
+            refund escrow ownPubKey
+    in void payAndRefund <|> void (redeemEp escrow)
```

Comment:
I find this slightly weird, but I guess it makes sense - I would expect to be racing `(awaitSlot >> refund) <|> redeem` rather than the whole `payAndRefund` thing, but I guess it does the same thing.

I guess this version will let you redeem before you pay if someone else makes your payment for you!